src/pages/Calendar/Calendar-FlowFilter.vue

```diff
@@ -0,0 +1,273 @@
+import { formatTime } from '@/mixins/formatTimeMixin'
+import FlowName from '@/pages/Calendar/FlowName'
+
+export default {
+  components: {
+    FlowName
+  },
+  mixins: [formatTime],
+  props: {
+    day: {
+      type: String,
+      required: true
+    }
+  },
+  data() {
+    return {
+      filters: [{ name: 'Flows' }],
+      skip: false,
+      show: true,
+      loadingKey: 0,
+      refetching: false,
+      Ids: null,
+      flowGroupIds: []
+    }
+  },
+  computed: {
+    ...mapGetters('api', ['isCloud']),
+    ...mapGetters('tenant', ['tenant']),
+    ...mapGetters('user', ['timezone']),
+    ...mapGetters('api', ['backend']),
+    start() {
+      let days = 1
+      const start =
+        this.tzOffset(this.day) < 0
+          ? this.subtractDay(this.day, days)
+          : this.day
+      return start
+    },
+    end() {
+      let days = 1
+      switch (this.type) {
+        case '4day':
+          days = this.tzOffset(this.day) > 0 ? 5 : 4
+          break
+        case 'day':
+          days = this.tzOffset(this.day) > 0 ? 2 : 1
+          break
+      }
+      return this.addDay(this.day, days)
+    },
+    allIds() {
+      const allRuns =
+        new Date(this.day) < new Date()
+          ? [
+              ...(this.flowRuns ? this.flowRuns : []),
+              ...(this.scheduledFlowRuns ? this.scheduledFlowRuns : []),
+              ...(this.runningFlowRuns ? this.runningFlowRuns : []),
+              ...(this.ongoingFlowRuns ? this.ongoingFlowRuns : [])
+            ]
+          : [
+              ...(this.flowRuns ? this.flowRuns : []),
+              ...(this.scheduledFlowRuns ? this.scheduledFlowRuns : [])
+            ]
+      const selected = this.selectFlow ? this.selectFlow[0] : null
+      const flowIds = allRuns?.reduce((accum, flowRun) => {
+        if (flowRun.flow_id !== selected) {
+          accum.push([flowRun.flow_id, 'active'])
+        }
+        return accum
+      }, [])
+      const flowGroupIds = this.allFlows?.reduce((accum, flowGroup) => {
+        if (flowGroup.flows[0]?.id && flowGroup.flows[0]?.id !== selected) {
+          accum.push([flowGroup.flows[0].id, `${flowGroup.flows[0].name}`])
+        }
+        return accum
+      }, [])
+      const allIds =
+        flowIds && flowGroupIds ? new Map([...flowGroupIds, ...flowIds]) : []
+      const runs = [...allIds]
+      const ordered = runs.sort((a, b) =>
+        a[1] === 'active'
+          ? -1
+          : b[1] === 'active'
+          ? 1
+          : a[1] > b[1]
+          ? 1
+          : b[1] > a[1]
+          ? -1
+          : 0
+      )
+      if (this.selectFlow) ordered.unshift(this.selectFlow)
+      return ordered
+    }
+  },
+  watch: {
+    async tenant() {
+      this.refetching = true
+      await this.$apollo.queries.flowRuns.refetch()
+      await this.$apollo.queries.scheduledFlowRuns.refetch()
+      await this.$apollo.queries.ongoingFlowRuns.refetch()
+      await this.$apollo.queries.runningFlowRuns.refetch()
```

Comment:
If we need to await these, let's collect them with `Promise.all` (in the `backend` watcher as well)